src/pykiso/interfaces/dt_auxiliary.py

```diff
@@ -0,0 +1,368 @@
+        self.stop_rx = threading.Event()
+        self.queue_in = queue.Queue()
+        self.queue_out = queue.Queue()
+        self.tx_task_on = tx_task_on
+        self.rx_task_on = rx_task_on
+        self.tx_thread = None
+        self.rx_thread = None
+        self.recv_timeout = 1
+        self.is_instance = False
+
+    @staticmethod
+    def initialize_loggers(loggers: Optional[List[str]]) -> None:
+        """Deactivate all external loggers except the specified ones.
+
+        :param loggers: list of logger names to keep activated
+        """
+        if loggers is None:
+            loggers = list()
+        # keyword 'all' should keep all loggers to the configured level
+        if "all" in loggers:
+            log.warning(
+                "All loggers are activated, this could lead to performance issues."
+            )
+            return
+        # keep package and auxiliary loggers
+        relevant_loggers = {
+            name: logger
+            for name, logger in logging.root.manager.loggerDict.items()
+            if not (name.startswith(PACKAGE) or name.endswith("auxiliary"))
+            and not isinstance(logger, logging.PlaceHolder)
+        }
+        # keep child loggers
+        childs = [
+            logger
+            for logger in relevant_loggers.keys()
+            for parent in loggers
+            if (logger.startswith(parent) or parent.startswith(logger))
+        ]
+        loggers += childs
+        # keep original level for specified loggers
+        loggers_to_deactivate = set(relevant_loggers) - set(loggers)
+        for logger_name in loggers_to_deactivate:
+            logging.getLogger(logger_name).setLevel(logging.WARNING)
+
+    def run_command(
+        self,
+        cmd_message: Any,
+        cmd_data: Any = None,
+        blocking: bool = True,
+        timeout_in_s: int = 5,
+    ) -> Any:
+        """Send a request by transmitting it through queue_in and
+        waiting for a response using queue_out.
+
+        :param cmd_message: command request to the auxiliary
+        :param cmd_data: data you would like to populate the command
+            with
+        :param blocking: If you want the command request to be
+            blocking or not
+        :param timeout_in_s: Number of time (in s) you want to wait
+            for an answer
+
+        :return: True if the request is correctly executed otherwise
+            False
+        """
+        with self.lock:
+            log.debug(
+                f"sending command '{cmd_message}' with payload {cmd_data} using {self.name} aux."
+            )
+            response_received = None
+            self.queue_in.put((cmd_message, cmd_data))
+            try:
+                response_received = self.queue_out.get(blocking, timeout_in_s)
+                log.debug(
+                    f"reply to command '{cmd_message}' received: '{response_received}' in {self.name}"
+                )
+            except queue.Empty:
```

Comment:
I'm a bot bothered by this right now. If I said `blocking=False`, why should I get an error? Maybe I do not want to wait for it. Than I somehow still get a log error message that will confuse me

Reply:
blocking=False , this mean if something is not directly in the queue the exception queueEmpty is raised so it advertise you that you expected soething directly but nothing is in the queue